compiler/src/dmd/expressionsem.d

```diff
@@ -9934,7 +9934,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             if (isArrayCtor || isArraySetCtor)
             {
                 const ts = t1b.nextOf().baseElemOf().isTypeStruct();
-                if (!ts || (!ts.sym.postblit && !ts.sym.dtor))
+                if (!ts || (!ts.sym.postblit && !ts.sym.hasCopyCtor && !ts.sym.dtor))
```

Comment:
This *almost* corresponds to `!ts || ts.isPOD()`. *Nested* structs make them non-POD too: not sure if those require the lowering too.

Reply:
IMO whether a struct is nested or not is orthogonal to the requirements for calling the hook. If a nested struct has a copy constructor then it should be called whenever an instance of that struct is created. However, the question of what to do with the context pointer still stands (https://issues.dlang.org/show_bug.cgi?id=22239); the solution would be to just copy it over to the destination, but that requires some new machinery in the compiler frontend (note that the IR is well equipped for such a task). Anyway, I digress.

Coming back to your comment, I think that `isPOD` is not the correct solution in this case.

Reply:
FWIW, I think the specialized logic (instead of using `isPOD()`) paved the way for the bug in the first place. - This seems like a very minor optimization - if a struct isn't a POD solely because of the context pointer, it's passed as-is to the glue layer without lowering, which blits the thing.

Maybe we should revise `isPOD()` wrt. context pointer - that making it a non-POD doesn't make too much sense OTOH, it's just an extra pointer and doesn't affect blittability of the thing.

Reply:
Makes sense to me.